src/jit/lower.cpp

```diff
@@ -628,20 +620,14 @@ GenTree* Lowering::LowerSwitch(GenTree* node)
 
                 // Now, build the conditional statement for the current case that is
                 // being evaluated:
-                // GT_JTRUE
-                //   |__ GT_COND
-                //          |____GT_EQ
-                //                 |____ (switchIndex) (The temp variable)
-                //                 |____ (ICon)        (The actual case constant)
-                GenTreePtr gtCaseCond =
-                    comp->gtNewOperNode(GT_EQ, TYP_INT, comp->gtNewLclvNode(tempLclNum, tempLclType),
-                                        comp->gtNewIconNode(i, TYP_INT));
-                /* Increment the lvRefCnt and lvRefCntWtd for temp */
+                // GT_CMP switchValue, caseValue
+                // GT_JCC(EQ)
```

Comment:
> This is a kind of transform that cannot be done properly during codegen. Since the immediate value is not contained a register is allocated to it and the mov is generated before we get to see the test node and do something about it.

Admittedly this would need to happen pre-RA.  We could probably still do it as long as we can tell that the def of the mov is anticipated (use in the test is last use) but that still leaves the RA in the lurch with extra register pressure.  Ideally this would be done in lower so the RA doesn't have to allocated the extra kill.